src/services/new/vm/vm-runner.ts

```diff
@@ -464,8 +464,12 @@ class VmRunner {
 
     let modules = {}
     for (let [contract_id, code] of Object.entries<string>(this.modules)) {
-      const binaryData = await ipfs.block.get(IPFS.CID.parse(code))
-      modules[contract_id] = await WebAssembly.compile(binaryData)
+      const binaryData = await ipfs.dag.get(IPFS.CID.parse(code))
+      try {
+        modules[contract_id] = await WebAssembly.compile(binaryData.value)
+      } catch (e) {
+        console.error(`invalid contract code ${contract_id}`, e)
```

Comment:
Throw and return error when unable to compile?

Reply:
I don't want to block other contract execution if there is only 1 bad contract

Reply:
Also, throwing an error here crashes the process and prevents block production. I do want to put a more robust solution in place to guarantee that all contracts are valid, but that will have to wait for the contract template refactoring work to be complete.

Reply:
I mean not throwing to crash, but return back to the contract output stating invalid contract code

Reply:
This is just a sanity check once the data availability proof is updated. Until then, this is the only thing stopping someone from halting the change again. If you still insist on a code change, please share it in a suggestion.

Reply:
We can leave it for now. Data availability check is sufficient.